pkg/images/utils/utils.go

```diff
@@ -44,9 +44,13 @@ func GenerateImageFromStringWithDefaultTag(imageStr, defaultTag string) (*storag
 		image.Id = digest.Digest().String()
 	}
 
-	// Default the image to latest if and only if there was no tag specific and also no SHA specified
-	if image.GetId() == "" && image.GetName().GetTag() == "" && defaultTag != "" {
-		SetImageTagNoSha(image.Name, defaultTag)
+	// Default the image tag to flavor if it wasn't set. Image tag shouldn't be empty
+	if image.GetName().GetTag() == "" && defaultTag != "" {
+		image.Name.Tag = defaultTag
+		// If an image SHA was provided we shouldn't try to update full name with the default tag
+		if image.Id == "" {
+			NormalizeImageFullNameNoSha(image.Name)
+		}
```

Comment:
In ROX-16212 case, `image.Id` is not empty. What do we get in result, `image:tag@sha256:digest`?